compiler/daml-lf-tools/src/DA/Daml/LF/TypeChecker/NameCollision.hs

```diff
@@ -107,41 +128,49 @@ fullyResolve = FRName . map T.toLower . \case
 -- | State of the name collision checker. This is a
 -- map from fully resolved names within a package to their
 -- original names. We update this map as we go along.
-newtype NCState = NCState (M.Map FRName [Name])
+data NCState = NCState (M.Map FRName [Name])
```

Comment:
What's wrong with the `newtype`? Not lazy enough?

Reply:
oops, that was leftover from an experiment where I tried to add two fields to this, reverted.